src/amo/components/OverallRating.js

```diff
@@ -70,25 +97,57 @@ export class OverallRatingBase extends React.Component {
   }
 }
 
-export const mapStateToProps = (state) => ({
-  apiState: state.api,
-  userId: state.auth && state.auth.userId,
-});
+export const mapStateToProps = (state, ownProps) => {
+  const userId = state.auth && state.auth.userId;
+  let userReview;
+
+  // Look for an already saved review by this user for this add-on/version.
+  if (userId && state.reviews) {
+    const allUserReviews = state.reviews[userId];
+    log.info(dedent`Checking state for review by user ${userId},
+      addonId ${ownProps.addonId}, versionId ${ownProps.version.id}`);
+    const addonReviews =
+      allUserReviews ? allUserReviews[ownProps.addonId] : null;
+
+    if (addonReviews) {
+      for (const reviewId of Object.keys(addonReviews)) {
+        const review = addonReviews[reviewId];
+        if (review.isLatest) {
+          userReview = review;
+          log.info('Found the latest review in state for this component',
+                   userReview);
+          break;
+        }
+      }
+    }
+  }
+
+  return {
+    apiState: state.api,
+    userReview,
+    userId,
+  };
+};
 
 export const mapDispatchToProps = (dispatch) => ({
-  createRating({ router, addonSlug, addonId, userId, ...params }) {
+
+  loadSavedRating({ userId, addonId }) {
+    return getUserReviews({ userId, addonId, onlyTheLatest: true })
+      .then((review) => {
+        if (review) {
+          dispatch(setReview(review));
+        } else {
+          log.info(
+            `No saved review found for userId ${userId}, addonId ${addonId}`);
+        }
+      });
+  },
+
+  submitRating({ router, addonSlug, ...params }) {
```

Comment:
Okay, I wasn't sure if that was the case. It probably won't be as bad here as on Marketplace. Rating might be a better term (although the API isn't using it) since reviewers review add-ons. Seems unlikely we'd get content ratings added to the mix like Marketplace had.